ui/components/auth-required.js

```diff
@@ -4,14 +4,21 @@ import useSWR from 'swr'
 export default function AuthRequired({ children }) {
   const { data: auth, error } = useSWR('/api/users/self')
   const router = useRouter()
+  const { pathname } = router
```

Comment:
We need to copy the entire path as a url to conserve query parameters. See `router.asPath`: https://nextjs.org/docs/api-reference/next/router#router-object